### PR TITLE
Add `HebrewCalendarHolidays` to holidays groups, refactor Israel holidays

### PR DESCRIPTION
**Pull request by KJhellico**
## Proposed change

Add `HebrewCalendarHolidays` to holidays groups (for use in other countries with Hebrew holidays); refactor Israel holidays.

## Type of change

- [ ] New country/market holidays support (thank you!)
- [ ] Supported country/market holidays update (calendar discrepancy fix, localization)
- [x] Existing code/documentation/test/process quality improvement (best practice, cleanup, refactoring, optimization)
- [ ] Dependency update (version deprecation/pin/upgrade)
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] Breaking change (a code change causing existing functionality to break)
- [ ] New feature (new `holidays` functionality in general)

## Checklist

- [x] I've followed the [contributing guidelines][contributing-guidelines]
- [x] I've successfully run `make check`, all checks and tests are green

[contributing-guidelines]: https://github.com/vacanza/holidays/blob/dev/CONTRIBUTING.rst
[docs]: https://github.com/vacanza/holidays/tree/dev/docs/source

### CODE REVIEW

**Review by arkid15r — COMMENTED** [review]
Yeah, this refactoring is long overdue. Thanks for addressing it!

File: holidays/groups/hebrew.py
Comment (arkid15r) [inline holidays/groups/hebrew.py:1]: I don't see the comment I thought I had left yesterday. Neither here nor in the mail. I didn't send it somehow -- sorry.

So the suggestion was about the other layer of abstraction we have for calendars. The methods like `def _yom_kippur_date():` should be there and in `groups/hebrew.py` you could operate w/ methods like `def _add_yom_kippur()`.

So in `israel.py` you could use `self._add_yom_kippur(tr("יום כיפור")) directly from the Hebrew holidays group.

Comment (KJhellico) [reply holidays/groups/hebrew.py:1]: Yes, initially I started with functions like `_add_yom_kippur()`. But in Israel, most holidays will require date value anyway (to handle with "observed" or add the Nth day of holiday).

Comment (PPsyrius) [reply holidays/groups/hebrew.py:1]: @KJhellico understandable, although I prefer @arkid15r's standardization approach more as that's what we've implemented for Islamic and Christian holiday groups.

Keep in mind that [Argentina also observes these officially as well](https://en.wikipedia.org/wiki/Public_holidays_in_Argentina) (Passover, Rosh Hashanah, and Yom Kippur)

Comment (KJhellico) [reply holidays/groups/hebrew.py:1]: Yes, I started with this when I started working on Montenegro.

**Review by PPsyrius — APPROVED** [review]
LGTM

**Review by arkid15r — COMMENTED** [review]
LGTM, thanks for keeping it well-structured.

**Pull request merged**